src/data_info/clients/async_tasks.clj

```diff
@@ -68,49 +65,47 @@
-           update-fn (fn [path action]
-                       (otel/with-span [s ["update-fn"]]
-                         (log/info "Updating async task:" async-task-id ":" path action)
-                         ;; we use the thread pool version here to be non-blocking, but elsewhere we use it so things happen in order (but deref the result to wait for the result before continuing)
-                         (retry-via-threadpool pool 3
-                           (fn [e t] (log/error (:throwable t) "failed updating async task"))
-                           add-status async-task-id {:status "running" :detail (format "[%s] %s: %s" (config/service-identifier) path (name action))})))]
-       (try+
-         (deref (retry-via-threadpool pool 3
-           (fn [e t] (log/error (:throwable t) "failed updating async task with started status"))
-           add-status async-task-id {:status "started"}))
-         (if use-client-user?
-           (irods/with-jargon-exceptions :client-user username [cm]
-             (otel/with-span [s ["jargon-fn" {:attributes {"client-user" username}}]]
-               (jargon-fn cm async-task update-fn)))
-           (irods/with-jargon-exceptions [cm]
-             (otel/with-span [s ["jargon-fn"]]
-               (jargon-fn cm async-task update-fn))))
-           ;; For the completed statuses we want a lot of retries because the presence or absence of an end date controls locking behavior
-           (deref (retry-via-threadpool pool 100
-             (fn [e t] (log/error (:throwable t) "failed updating async task with completed status"))
-             add-completed-status async-task-id {:status "completed" :detail (str "[" (config/service-identifier) "]")}))
-         (otel/with-span [s ["end-fn"]]
-           (end-fn async-task false))
-         (catch Object _
-           (log/error (:throwable &throw-context) "failed processing async task" async-task-id)
-           (deref (retry-via-threadpool pool 100
-             (fn [e t] (log/error (:throwable t) "failed updating async task with completed status"))
-             add-completed-status async-task-id {:status "failed" :detail (format "[%s] %s" (config/service-identifier) (pr-str (:throwable &throw-context)))}))
-           (end-fn async-task true))))
-     (log/info "Finished processing async task " async-task-id))))
+   (let [pool                              (make-threadpool (str async-task-id "-async-tasks-update") 1)
+         {:keys [username] :as async-task} (get-by-id async-task-id)
+         update-fn                         (create-update-fn async-task-id pool)]
+     (try+
+      (deref (retry-via-threadpool pool 3
+                                   (fn [e t] (log/error (:throwable t) "failed updating async task with started status"))
+                                   add-status async-task-id {:status "started"}))
+      (if use-client-user?
+        (irods/with-jargon-exceptions :client-user username [cm] jargon-fn)
```

Comment:
This looks off to me -- shouldn't `jargon-fn` be `(jargon-fn cm async-task update-fn)` here? Otherwise I guess it'd just return the function, not run it.

Reply:
Good catch! I was relying on my editor to update the code a little too much, and I must have had the cursor in the wrong place when I did it. I'll fix it. 👍